src/Bigcommerce/Api/Connection.php

```diff
@@ -8,455 +8,469 @@
-		curl_setopt($this->curl, CURLOPT_URL, $url);
-		curl_setopt($this->curl, CURLOPT_NOBODY, true);
-		curl_exec($this->curl);
-
-		return $this->handleResponse();
-	}
-
-	/**
-	 * Make an HTTP PUT request to the specified endpoint.
-	 *
-	 * Requires a tmpfile() handle to be opened on the system, as the cURL
-	 * API requires it to send data.
-	 */
-	public function put($url, $body)
-	{
-		$this->addHeader('Content-Type', $this->getContentType());
-
-		if (!is_string($body)) {
-			$body = json_encode($body);
-		}
-
-		$this->initializeRequest();
-
-		$handle = tmpfile();
-		fwrite($handle, $body);
-		fseek($handle, 0);
-		curl_setopt($this->curl, CURLOPT_INFILE, $handle);
-		curl_setopt($this->curl, CURLOPT_INFILESIZE, strlen($body));
-
-		curl_setopt($this->curl, CURLOPT_CUSTOMREQUEST, 'PUT');
-		curl_setopt($this->curl, CURLOPT_URL, $url);
-		curl_setopt($this->curl, CURLOPT_PUT, true);
-		curl_exec($this->curl);
-
-		return $this->handleResponse();
-	}
-
-	/**
-	 * Make an HTTP DELETE request to the specified endpoint.
-	 */
-	public function delete($url)
-	{
-		$this->initializeRequest();
-
-		curl_setopt($this->curl, CURLOPT_CUSTOMREQUEST, 'DELETE');
-		curl_setopt($this->curl, CURLOPT_URL, $url);
-		curl_exec($this->curl);
-
-		return $this->handleResponse();
-	}
-
-	/**
-	 * Callback method collects body content from the response.
-	 */
-	private function parseBody($curl, $body)
-	{
-		$this->responseBody .= $body;
-		return strlen($body);
-	}
-
-	/**
-	 * Callback methods collects header lines from the response.
-	 */
-	private function parseHeader($curl, $headers)
-	{
-		if (!$this->responseStatusLine && strpos($headers, 'HTTP/') === 0) {
-			$this->responseStatusLine = $headers;
-		} else {
-			$parts = explode(': ', $headers);
-			if (isset($parts[1])) {
-				$this->responseHeaders[$parts[0]] = trim($parts[1]);
-			}
-		}
+    /**
+     * @var cURL resource
+     */
+    private $curl;
+
+    /**
+     * @var hash of HTTP request headers
+     */
+    private $headers = array();
+
+    /**
+     * @var hash of headers from HTTP response
+     */
+    private $responseHeaders = array();
+
+    /**
+     * The status line of the response.
+     * @var string
+     */
+    private $responseStatusLine;
+
+    /**
+     * @var hash of headers from HTTP response
+     */
+    private $responseBody;
+
+    /**
+     * @var boolean
+     */
+    private $failOnError = false;
+
+    /**
+     * Manually follow location redirects. Used if CURLOPT_FOLLOWLOCATION
+     * is unavailable due to open_basedir restriction.
+     * @var boolean
+     */
+    private $followLocation = false;
+
+    /**
+     * Maximum number of redirects to try.
+     * @var int
+     */
+    private $maxRedirects = 20;
+
+    /**
+     * Number of redirects followed in a loop.
+     * @var int
+     */
+    private $redirectsFollowed = 0;
+
+    /**
+     * Deal with failed requests if failOnError is not set.
+     * @var mixed
```

Comment:
Instead of `mixed`, could we say `string | false` or something like that? (I know `false` doesn't seem like a type, but it is what phpdoc suggests.)